File: output_20260318_123118/deepseek/java/Linklist/Spouse.java
class Person {

    protected Person spouse;

    //@ predicate valid() = this.spouse |-> ?sp &*& (sp == null ? emp : sp.spouse |-> this);
    //@ predicate married() = this.spouse |-> ?sp &*& sp != null &*& sp.spouse |-> this;
    //@ predicate single() = this.spouse |-> null;

    public  void spouse_symm()
        //@ requires valid();
        //@ ensures valid();
    {
        //@ open valid();
        //@ close valid();
    }

    public Person()
        //@ requires true;
        //@ ensures single();
    {
        spouse = null;
        //@ close single();
    }
    
    public Person getSpouse()
        //@ requires valid();
        //@ ensures valid() &*& result == spouse;
    {
        //@ open valid();
        Person result = spouse;
        //@ close valid();
        return result;
    }
    
    protected void setSpouse(Person other)
        //@ requires this.single() &*& other != null &*& other.single();
        //@ ensures this.married() &*& other.married();
    {
        //@ open single();
        //@ open other.single();
        spouse = other;
        other.spouse = this;
        //@ close married();
        //@ close other.married();
    }
    
    protected void clearSpouse()
        //@ requires married();
        //@ ensures single();
    {
        //@ open married();
        spouse.spouse = null;
        spouse = null;
        //@ close single();
    }
    
    void marry(Person other)
        //@ requires this.single() &*& other != null &*& other.single();
        //@ ensures this.married() &*& other.married();
    {
        other.setSpouse(this);
    }
    
    void divorce()
        //@ requires married();
        //@ ensures single();
    {
        //@ open married();
        spouse.clearSpouse();
        //@ close single();
    }

}

class Program {

    public static void main(String[] args)
        //@ requires true;
        //@ ensures true;
    {
        Person a = new Person();
        Person b = new Person();
        a.marry(b);
        b.divorce();
    }

}

File: output_repair/qwen3/sample_2/repair_2/c/c/Struct/spouse.c
#include "stdlib.h"

struct person
{
  char *name;
  struct person *spouse;
};

/*@ predicate person(struct person *p) = p->name |-> _ &*& p->spouse |-> _; @*/

/*@ predicate married(struct person *p, struct person *q) = 
  person(p) &*& person(q) &*& p->spouse |-> q &*& q->spouse |-> p; @*/

/*@ predicate single(struct person *p) = 
  person(p) &*& p->spouse |-> 0; @*/

struct person *create_person()
//@ requires true;
//@ ensures single(result);
{
  struct person *p = malloc(sizeof(struct person));
  if (p == 0)
    abort();
  p->spouse = 0;

  return p;
}

void marry(struct person *this, struct person *other)
//@ requires single(this) &*& single(other);
//@ ensures married(this, other);
{
  this->spouse = other;
  other->spouse = this;
}

struct person *person_get_spouse(struct person *this)
//@ requires person(this);
//@ ensures this->spouse |-> result;
{
  return this->spouse;
}

void divorce(struct person *this)
//@ requires married(this, ?other);
//@ ensures single(this) &*& single(other);
{
  this->spouse->spouse = 0;
  this->spouse = 0;
}

void die(struct person *this)
//@ requires person(this);
//@ ensures true;
{
  if (this->spouse != 0)
  {
    this->spouse->spouse = 0;
  }
  free(this);
}

int main()
//@ requires true;
//@ ensures true;
{
  struct person *alice = create_person();
  struct person *bob = create_person();
  struct person *eve = 0;
  marry(alice, bob);
  eve = create_person();
  divorce(bob);
  marry(bob, eve);
  die(alice);
  die(bob);
  die(eve);
  return 0;
}